app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $areaSqDegrees = DB::selectOne('SELECT ST_Area(geom) AS area FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->area;
+        $latitude = DB::selectOne('SELECT ST_Y(ST_Centroid(geom)) AS latitude FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->latitude;
+        $areaSqMeters = $areaSqDegrees * pow(111320 * cos(deg2rad($latitude)), 2);
+        $SIZE_CRITERIA_ID = 6;
+        $valid = $areaSqMeters <= 10000000;
+        $insertionSuccess = $this->insertCriteriaSite($uuid, $SIZE_CRITERIA_ID, $valid);
+
+        return response()->json([
+          'area_hectares' => $areaSqMeters / 10000, // Convert to hectares
+          'area_sqmeters' => $areaSqMeters,
+          'geometry_id' => $geometry->id,
+          'insertion_success' => $insertionSuccess,
+          'valid' => $valid,
+        ], 200);
+    }
+
+    public function checkWithinCountry(Request $request)
+    {
+        $polygonUuid = $request->input('uuid');
+
+        if ($polygonUuid === null || $polygonUuid === '') {
+            return response()->json(['error' => 'UUID not provided'], 200);
+        }
+
+        $geometry = PolygonGeometry::where('uuid', $polygonUuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $totalArea = DB::table('polygon_geometry')
+            ->where('uuid', $polygonUuid)
+            ->selectRaw('ST_Area(geom) AS area')
+            ->first()->area;
+
+        // Find site_polygon_id and project_id using the polygonUuid
+        $sitePolygonData = SitePolygon::where('poly_id', $polygonUuid)
+            ->select('id', 'project_id')
+            ->first();
+
+        if (! $sitePolygonData) {
+            return response()->json(['error' => 'Site polygon data not found for the specified polygonUuid'], 404);
+        }
+
+        // Find the country ISO using project_id from v2projects
+        $countryIso = DB::table('v2_projects')
+            ->where('uuid', $sitePolygonData->project_id)
+            ->value('country');
+
+        if (! $countryIso) {
+            return response()->json(['error' => 'Country ISO not found for the specified project_id'], 404);
+        }
+
+        $intersectionData = DB::table('world_countries_generalized')
+          ->where('iso', $countryIso)
+          ->selectRaw('world_countries_generalized.country AS country, ST_Area(ST_Intersection(world_countries_generalized.geometry, (SELECT geom FROM polygon_geometry WHERE uuid = ?))) AS area', [$polygonUuid])
+          ->first();
+
+        $intersectionArea = $intersectionData->area;
+        $countryName = $intersectionData->country;
+
+        $insidePercentage = $intersectionArea / $totalArea * 100;
+
+        $insideThreshold = 75;
+        $insideViolation = $insidePercentage < $insideThreshold;
+        $WITHIN_COUNTRY_CRITERIA_ID = 7;
```

Comment:
I'll stop commenting on these now, but I'm super curious why:
* All these magic numbers are sprinkled around?
* If they really are necessary, they should be specified as constants in the controller, or if used in other code, in a shared location.

Reply:
Resolving this as the conversation is continuing in an earlier comment.